lldb/source/Symbol/SaveCoreOptions.cpp

```diff
@@ -46,8 +46,59 @@ SaveCoreOptions::GetOutputFile() const {
   return m_file;
 }
 
+void SaveCoreOptions::AddThread(lldb::tid_t tid) {
+  if (m_threads_to_save.count(tid) == 0)
+    m_threads_to_save.emplace(tid);
+}
+
+bool SaveCoreOptions::RemoveThread(lldb::tid_t tid) {
+  if (m_threads_to_save.count(tid) == 0) {
+    m_threads_to_save.erase(tid);
+    return true;
+  }
```

Comment:
FYI this would fail to remove the tid because you have `m_threads_to_save.count(tid) == 0` instead of `m_threads_to_save.count(tid) != 0`, but see the one line replacement code that will work below.